cdb2api/cdb2api.c

```diff
@@ -2348,13 +2348,15 @@ static int cdb2_send_query(cdb2_hndl_tp *hndl, SBUF2 *sb, const char *dbname,
     gettimeofday(&tv, NULL);
     sqlquery.timestampus = ((uint64_t)tv.tv_sec) * 1000000 + tv.tv_usec;
 
+    sqlquery.has_num_retries = 1;
+    sqlquery.num_retries = retries_done;
+
```

Comment:
https://github.com/bloomberg/comdb2/blob/469098e0eea46e555a170d4db65bf08a3664ea79/cdb2api/cdb2api.c#L2281

Reply:
https://github.com/bloomberg/comdb2/blob/469098e0eea46e555a170d4db65bf08a3664ea79/cdb2api/cdb2api.c#L3141

Reply:
Mohit, i'm a bit confused, I thought there is the api retrying on the same node upon receiving rc as in is_retryable() function. Then there is the retry against a different node upon failure to get response from the server on timeout or if server has no threads available to process (we don't put in queue yet). Do we use same counter for both? 
Is retries_done not the second one?  
```
2301         if (retries_done >= hndl->num_hosts) {                                                                                                    
2302             features[n_features] = CDB2_CLIENT_FEATURES__ALLOW_QUEUING;
2303             n_features++;
2304         }
```
Thanks.

Reply:
In my testing, if i randomly return CDB2__ERROR_CODE__CHANGENODE from server (in handle_newsql_request()) I don't see retry ever being nonzero because has_is_retry is never set!! Anyway, i see my num_retry often being nonzero, meaning when I change server it reports counter increased. Not sure which one to use yet.